### PR TITLE
hw3 2_MiniLogo

### PR DESCRIPTION
**Pull request by psulliv**
```
>./2_MiniLogo 
./2_MiniLogo 
"Tests for hw3 2_mini_logo"
"Running this sample mini logo programe"
Sequ (Sequ (Sequ (Sequ (Sequ (Pen Down) (MoveTo 1 1)) (MoveTo 0 1)) (MoveTo 0 0)) (Pen Up)) (MoveTo 50 50)
[(0,0,1,1),(1,1,0,1),(0,1,0,0)]
```

### CODE REVIEW

**Review by osu-mp — APPROVED** [review]
Looks great, way to go!

File: hw3/2_MiniLogo.hs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+type State = (Mode, Int, Int)
+
+type Line = (Int, Int, Int, Int)
+
+type Lines = [Line]
+
+semL :: Cmd -> State -> (State, Lines)
+semL (Pen m) (_, x, y) = ((m, x, y), [])
+semL (MoveTo a b) (Down, x, y) = ((Down, a, b), [(x, y, a, b)])
+semL (MoveTo a b) (Up, x, y) = ((Down, x, y), [])
```
Comment (caitlynplewis) [inline hw3/2_MiniLogo.hs]: If I move the pen while it's up, why would it then become down after I finish the move? Shouldn't it stay up?

**Review by caitlynplewis — APPROVED** [review]
Looks good! I think there's one quirk with your semL that should be fixed but otherwise great.

File: hw3/2_MiniLogo.hs (hunk 1 above)
Comment (psulliv) [reply hw3/2_MiniLogo.hs]: good catch! I'll change that.

**Comment by psulliv** [comment]
Fixed bug:
```
>./2_MiniLogo 
./2_MiniLogo 
"Tests for hw3 2_mini_logo"
"Running this sample mini logo program:"
Sequ (Sequ (Sequ (Sequ (Sequ (Pen Down) (MoveTo 1 1)) (MoveTo 0 1)) (MoveTo 0 0)) (Pen Up)) (MoveTo 50 50)
"We output these lines:"
[(0,0,1,1),(1,1,0,1),(0,1,0,0)
```

**Pull request merged**